### ISSUE TITLE
iOS re-desing proposal 

### ISSUE THREAD

**Issue by robagreda** [damus-io/damus#82]
Hello mates, contacted jb on Damus via DM, and shared this re-design version of the Damus app for iOS, he liked it already but he wants me to share it with you all, to see what you all think and if you have feedback or general comments.

Here is the Figma prototype view(now just scroll with blur effect): https://www.figma.com/proto/G4n8Fh3CNgU3Znu2Uhdt1W/Untitled?page-id=0%3A1&node-id=1%3A2&viewport=1361%2C433%2C0.53&scaling=scale-down

![image](https://user-images.githubusercontent.com/3373850/208946067-9930be85-c5d2-4751-a7a7-c127a6854e71.png)

If you all like, we can move ahead and work on DM, Notifications, Posts, and more. 

Thank you!

**Comment by jb55** [comment]
cc @sdub18 @ismyhc

**Comment by sdub18** [comment]
Oh yeah we can make that happen

**Comment by sdub18** [comment]
That timeline is gorgeous. We should for sure do a meetup sometime this week or something. Start making some actionable steps

**Comment by bebopkid** [comment]
thoughts on pubkey in the home/global feed? it has name and username but neither are actually unique. but design looks great!

**Comment by sdub18** [comment]
Personally I would be against that. For developers like us that is helpful but for regular people seeing the whole pubkey can definitely be jarring. Something I did on the Profile view was move that into an icon and then use pubkey_to_rgb_hex to give it a unique color based on value. That might be a bit user friendly. I definitely see where you are coming from though

**Comment by robagreda** [comment]
I see the little icon, works nicely to copy the key, would be nice to have the icon and once users touch it, raise a modal showing the public key, explain what that is a copy to clipboard button. Thoughts?

**Comment by sdub18** [comment]
Yeah I agree. That could be a nice middle ground. Definitely something to discuss further I think.

**Comment by sdub18** [comment]
I think the tab bar could also use some work. That middle `+` button is not calling to me personally. But overall I do like that concept. It just feels more React Native/Kotlin and less native SwiftUI if that makes any sense. Just my personal opinion so feel free to take with a grain of salt

**Comment by robagreda** [comment]
Hey, yes, will keep exploring more concepts about that! I know that the + to create a new post is super important and maybe the placement is not sooo good at this point, so thanks for the feedback, always appeaciated.

**Comment by bebopkid** [comment]
> I definitely see where you are coming from though

yeah def don't mean the whole pubkey 😅 just some way to help identify authenticity. i like the color idea but not sure it's easy to understand what it conveys immediately; however, either way, yes, some way to help ppl understand who's a fake jack and a real jack.

**Comment by niteshbalusu11** [comment]
Wow, looks super cool!!

**Comment by markrachapoom** [comment]
looking good! would love to contribute in the near future :))

**Comment by jakubszwedo** [comment]
Great stuff, If you want a hand I am happy to design with you.

**Comment by radixrat** [comment]
Small comment, Is it possible we could leave retweet the same icon as twitter? Think less confusion and no use reinventing what people already know. Mastodon clients drove me nuts because they change small things like this.

**Comment by sdub18** [comment]
> Small comment, Is it possible we could leave retweet the same icon as twitter? Think less confusion and no use reinventing what people already know. Mastodon clients drove me nuts because they change small things like this.

Haha thats so funny cuz I was feeling the exact opposite. But I totally agree with where you are coming from

**Comment by nvk** [comment]
Looks very pretty, my only suggestion would be to make information more dense. Maybe reduce the space between two posts and remove the button decoration around the icons. jm2cc

**Comment by jpye-finch** [comment]
dig it! would just check the accessibility on the purple links

**Comment by robagreda** [comment]
I can work on it, and change the colors, but I like them cuz you see right away the touch targets! :)

**Comment by nvk** [comment]
Another comment regarding the "Name" and "@name", nostr doesn't have @ native. It's is a pubkey, so i'd suggest:

- when nip5 available: "@nickname" • "nick@domain.tld"
- when not available: "@nickname" • "truncated pubkey"

**Comment by aerolit** [comment]
Looking nice already! 
Why not show the pics in full screenwide and blend the text/username over in a semi transparent ‚bubble‘ or something.

**Comment by scoder1747** [comment]
Can we also enumerate the Reply Icon so that the user knows how many replies have been already posted.
<img width="157" alt="Screen Shot 2022-12-21 at 1 56 06 PM" src="https://user-images.githubusercontent.com/120697811/208982186-fee2e55c-0f8a-4b11-89f6-e38d9a9d7429.png">

**Comment by scoder1747** [comment]
Also, we need to have a design for whether the post is a reply or a main post.

**Comment by scoder1747** [comment]
> Another comment regarding the "Name" and "@name", nostr doesn't have @ native. It is a pubkey, so i'd suggest:
> 
> * when nip5 available: "@Nickname" • "[nick@domain.tld](mailto:nick@domain.tld)" or "[nick@domain.tld](mailto:nick@domain.tld)" • "truncated pubkey"
> * when not available: "@Nickname" • "truncated pubkey"

Yeah I agree. Probably, we can use last 10 characters of pubkey for "truncated pubkey" (just like 5 digits of cr card).

**Comment by ismyhc** [comment]
Design looks great! However, Im more of a fan of using standard platform design especially in an open-sourced project as it makes it easier to maintain.

That said, it's a really great design and I have no real complaints outside of what's already been said. As long there are contributors willing to make these changes I think it's great!

Since there have been a few people step into help, I think Im going to focus on my DM focused app Seer! :)

**Comment by alixanc** [comment]
> > Another comment regarding the "Name" and "@name", nostr doesn't have @ native. It is a pubkey, so i'd suggest:
> > 
> > * when nip5 available: "@Nickname" • "[nick@domain.tld](mailto:nick@domain.tld)" or "[nick@domain.tld](mailto:nick@domain.tld)" • "truncated pubkey"
> > * when not available: "@Nickname" • "truncated pubkey"
> 
> Yeah I agree. Probably, we can use last 10 characters of pubkey for "truncated pubkey" (just like last 4 digits of cr card or ssn).

Couldn't that still be brute-forced by fake accounts, making them actually looking more legit instead of helping to identify the real accounts?

What about having a small "followers you follow" count directly visible, if you see an account is being followed by many of your follows that could be a reasonably secure indicator? Basically already what I'm using on Twitter, but more convenient if visible in timeline.

**Comment by robagreda** [comment]
Will be sharing a newer version soon, taking all your comments/feedback in consideration. You'll see more stuff as well so you can comment about. Thinking on adding a figma link so you can provide feedback right there! 😳🤔

**Comment by bitcoinuser** [comment]
I would like to see the "replying to" part. I think maybe instead of show name, could show the profile image. Like "replying to: profile image 1 profile image 2, show more."

**Comment by bitcoinuser** [comment]
I would use a different font for the app. Something more modern, like this:

![Screenshot_2022-12-21-23-17-29-172_com brave browser](https://user-images.githubusercontent.com/1178856/209040830-39296a61-c43d-4f10-86a1-db8bb1cc2924.jpg)

**Comment by robagreda** [comment]
I already changed the font from Inter to Satoshi(obviously) to honor Satoshi Nakamoto, en looks good! And also have included the Replying to UI. As well I've incorporated the pubkey truncated pill into profiles and posts. 

![image](https://user-images.githubusercontent.com/3373850/209041017-f096bb03-342f-4dcc-8a20-a6151ba20b8e.png)

![image](https://user-images.githubusercontent.com/3373850/209041498-6b0add92-7bcd-4b27-baa4-35911b2d3741.png)

**Comment by Bosch-0** [comment]
Very nice! Fellow product designer here, would love to help out. As you mentioned above sharing your Figma file (at least for commenting) would be handy and aligns with the whole open-source ethos. Maybe publish it as a Figma Community file so people can remix it?

As some others mentioned I would aim to keep things as close to Twitter as possible without copying it pixel by pixel. [Jakob's Law](https://lawsofux.com/en/jakobs-law/) applies here, people don't like change. Signal did a great job with their UX being quite similar to WhatsApp to make the friction of switching as minimal as possible. On the topic of Signal, rather then clutter the feeds with truncated pub keys we should do something similar to their "View Safety Number" feature where you click on the users profile and verify them. Maybe once you verify them it adds a small badge next to their name (similar to a Twitter blue badge) to show you have verified this person as someone you know. See below for what I am talking about:

![image](https://user-images.githubusercontent.com/55287964/209249680-6739c13e-e182-45d4-ad89-18d1624e363c.png)

**Comment by jakubszwedo** [comment]
@Bosch-0 100% agree on both Twitter-like design and opensourcing the design :)

The main difference with twitter would be Global and I'd think of adding chats in the future can be beneficial, so Damus would be a nostr-powerhouse, in contrary to some more niche/narrowed applications

**Comment by CH-SYR3** [comment]
This looks awesome! Would love to see this happen.

**Comment by alltheseas** [comment]
Closing as part of a backlog triage — we're narrowing focus to the most actionable issues to reduce noise for contributors.

If this is still relevant, please reopen with updated context or reproduction steps. Thanks for reporting!
